src/components/GitOpsDetailsPageTabs.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+import { RouteComponentProps } from 'react-router';
+
+import { HorizontalNav, NavPage } from '@openshift-console/dynamic-plugin-sdk';
+import { LoadingBox } from '@patternfly/quickstarts';
+
+import GitOpsDetailsPageHeading from './details/GitOpsDetailsPageHeading';
+import GitOpsDetailsPage from './GitOpsDetailsPage';
+import GitOpsDeploymentHistory from './history/GitOpsDeploymentHistory';
+import DevPreviewBadge from './import/badges/DevPreviewBadge';
+import { getPipelinesBaseURI } from './utils/gitops-utils';
+import useDefaultSecret from './utils/useDefaultSecret';
+import useEnvDetails from './utils/useEnvDetails';
+
+type GitOpsDetailsPageTabsProps = RouteComponentProps<{ appName?: string }>;
+
+export const GitOpsDetailsPageTabs: React.FC<GitOpsDetailsPageTabsProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const { appName } = match.params;
+  const [secretNS, secretName] = useDefaultSecret();
+  const pipelinesBaseURI = getPipelinesBaseURI(secretNS, secretName);
+  const searchParams = new URLSearchParams(location.search);
+  const manifestURL = searchParams.get('url');
+  const [envs, emptyStateMsg] = useEnvDetails(appName, manifestURL, pipelinesBaseURI);
+
+  const pages: NavPage[] = [
+    {
+      href: `${'overview?url='}${manifestURL}`,
+      name: t('gitops-plugin~Overview'),
+      path: 'overview',
+      component: GitOpsDetailsPage,
+    },
+    {
+      href: `${'deploymenthistory?url='}${manifestURL}`,
+      name: t('gitops-plugin~Deployment history'),
+      path: 'deploymenthistory',
+      component: GitOpsDeploymentHistory,
+    },
+  ];
+
+  return (
+    <>
+      <Helmet>
+        <title>{t('gitops-plugin~{{appName}} · Details', { appName })}</title>
+      </Helmet>
+      <GitOpsDetailsPageHeading
+        url={match.url}
+        appName={appName}
+        manifestURL={manifestURL}
+        badge={<DevPreviewBadge />}
+      />
+      {!emptyStateMsg && !envs ? <LoadingBox /> : <HorizontalNav pages={pages} />}
```

Comment:
Hi Yi, could you explain your solution for this?  In the static plugin, the data/props like applicationBaseURI, manifestURL, etc. were taken from GitOpsDetailsPageTabs.  Is it that the HorizontalNavProps no longer takes customData ?

Reply:
Yes, `HorizontalNavProps` doesn't allow to pass customData, do you have better solution for this?

Reply:
Problem resolved, thanks for the help 👍